src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -63,16 +61,19 @@ export const TradeConfirm = ({ history }: RouterProps) => {
     selectFirstAccountSpecifierByChainId(state, buyAssetChainId),
   )
 
-  const parsedTxId = useMemo(
-    () => serializeTxIndex(buyAssetAccountSpecifier, txid, trade.receiveAddress),
-    [buyAssetAccountSpecifier, trade.receiveAddress, txid],
+  const parsedBuyTxId = useMemo(
+    () => serializeTxIndex(buyAssetAccountSpecifier, buyTxid, trade.receiveAddress),
+    [buyAssetAccountSpecifier, trade.receiveAddress, buyTxid],
   )
-  const status = useAppSelector(state => selectTxStatusById(state, parsedTxId))
+
+  const status =
+    useAppSelector(state => selectTxStatusById(state, parsedBuyTxId)) ?? TxStatus.Pending
 
   const { showErrorToast } = useErrorHandler()
 
   const onSubmit = async () => {
     try {
+      setSellTxid('pending')
```

Comment:
we need to set it to something so it hides the button once they click it. because onSumit can block